<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SoccerControlFrame</class>
 <widget class="AttachableFrame" name="SoccerControlFrame">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>297</width>
    <height>565</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Frame</string>
  </property>
  <property name="whatsThis">
   <string>The soccer control frame can be used to monitor and control the state of a simspark soccer simulation.</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_3">
   <property name="margin">
    <number>0</number>
   </property>
   <item row="0" column="0">
    <widget class="QWidget" name="widget_6" native="true">
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="title">
         <string>Game State</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QWidget" name="widget_8" native="true">
           <property name="font">
            <font>
             <pointsize>8</pointsize>
            </font>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <property name="margin">
             <number>0</number>
            </property>
            <item>
             <widget class="QLabel" name="gameTimeHeaderLabel">
              <property name="font">
               <font>
                <pointsize>8</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Game Time:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="gameTimeEdit">
              <property name="sizePolicy">
               <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>100</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>100</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="toolTip">
               <string>Current game time.</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="gameTimeResetButton">
              <property name="toolTip">
               <string>Reset the game time.</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="../../imageresources.qrc">
                <normaloff>:/arrow_undo</normaloff>:/arrow_undo</iconset>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_2">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>59</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="Line" name="line_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="widget_3" native="true">
           <property name="font">
            <font>
             <pointsize>8</pointsize>
            </font>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <property name="margin">
             <number>0</number>
            </property>
            <item>
             <widget class="QWidget" name="widget" native="true">
              <property name="font">
               <font>
                <pointsize>8</pointsize>
               </font>
              </property>
              <layout class="QFormLayout" name="formLayout">
               <property name="verticalSpacing">
                <number>3</number>
               </property>
               <property name="margin">
                <number>0</number>
               </property>
               <item row="0" column="0">
                <widget class="QLabel" name="team1NameHeaderLabel">
                 <property name="font">
                  <font>
                   <pointsize>10</pointsize>
                   <weight>75</weight>
                   <bold>true</bold>
                  </font>
                 </property>
                 <property name="text">
                  <string>Team:</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QLabel" name="team1NameLabel">
                 <property name="text">
                  <string>TextLabel</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="team1GoalsHeaderLabel">
                 <property name="text">
                  <string>Goals:</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QLabel" name="team1PenaltiesHeaderLabel">
                 <property name="text">
                  <string>Penalties:</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QLineEdit" name="team1GoalsEdit">
                 <property name="toolTip">
                  <string>Golad of first team.</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <widget class="QLineEdit" name="team1PenaltiesEdit">
                 <property name="toolTip">
                  <string>Penalties of first team.</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QWidget" name="widget_2" native="true">
              <layout class="QFormLayout" name="formLayout_2">
               <property name="verticalSpacing">
                <number>3</number>
               </property>
               <property name="margin">
                <number>0</number>
               </property>
               <item row="0" column="0">
                <widget class="QLabel" name="team2NameHeaderLabel">
                 <property name="font">
                  <font>
                   <pointsize>10</pointsize>
                   <weight>75</weight>
                   <bold>true</bold>
                  </font>
                 </property>
                 <property name="text">
                  <string>Team:</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QLabel" name="team2NameLabel">
                 <property name="text">
                  <string>TextLabel</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="team2GoalsHeaderLabel">
                 <property name="text">
                  <string>Goals:</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QLabel" name="team2PenaltiesHeaderLabel">
                 <property name="text">
                  <string>Penalties:</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QLineEdit" name="team2GoalsEdit">
                 <property name="toolTip">
                  <string>Goals of second team.</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <widget class="QLineEdit" name="team2PenaltiesEdit">
                 <property name="toolTip">
                  <string>Penalties of second team.</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="Line" name="line">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="widget_4" native="true">
           <property name="font">
            <font>
             <pointsize>8</pointsize>
            </font>
           </property>
           <layout class="QGridLayout" name="gridLayout_4">
            <property name="margin">
             <number>0</number>
            </property>
            <item row="1" column="3">
             <widget class="QToolButton" name="playModeButton">
              <property name="toolTip">
               <string>Change the play mode.</string>
              </property>
              <property name="whatsThis">
               <string>Changes the current play mode to what is selected in the selection box.</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="../../imageresources.qrc">
                <normaloff>:/accept</normaloff>:/accept</iconset>
              </property>
             </widget>
            </item>
            <item row="1" column="2">
             <widget class="QComboBox" name="playModeComboBox">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="toolTip">
               <string>Change the play mode.</string>
              </property>
              <property name="whatsThis">
               <string>Choose a play mode here and activate it with the button.</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLabel" name="playModeHeaderLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>PlayMode:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QLabel" name="playModeLabel">
              <property name="text">
               <string>TextLabel</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="Line" name="line_3">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="widget_5" native="true">
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <property name="spacing">
             <number>4</number>
            </property>
            <property name="margin">
             <number>0</number>
            </property>
            <item>
             <widget class="QToolButton" name="kickOffButton">
              <property name="toolTip">
               <string>Kickoff</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="../../imageresources.qrc">
                <normaloff>:/sport_soccer</normaloff>:/sport_soccer</iconset>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="title">
         <string>Players</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QWidget" name="widget_7" native="true">
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <property name="margin">
             <number>0</number>
            </property>
            <item>
             <widget class="QListWidget" name="playersListWidget">
              <property name="font">
               <font>
                <pointsize>8</pointsize>
               </font>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QWidget" name="widget_9" native="true">
              <property name="font">
               <font>
                <pointsize>8</pointsize>
               </font>
              </property>
              <layout class="QGridLayout" name="gridLayout">
               <property name="margin">
                <number>0</number>
               </property>
               <item row="0" column="0">
                <widget class="QLabel" name="agentNameHeaderLabel">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>Name:</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="2">
                <widget class="QLabel" name="agentTeamHeaderLabel">
                 <property name="text">
                  <string>Team:</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="3">
                <widget class="QLabel" name="agentTeamLabel">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>TextLabel</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QLabel" name="agentNameLabel">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>TextLabel</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="widget_10" native="true">
           <property name="font">
            <font>
             <pointsize>8</pointsize>
            </font>
           </property>
           <layout class="QGridLayout" name="gridLayout_2">
            <property name="verticalSpacing">
             <number>2</number>
            </property>
            <property name="margin">
             <number>0</number>
            </property>
            <item row="0" column="0">
             <widget class="QLabel" name="batteryHeaderLabel">
              <property name="text">
               <string>Battery:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="3">
             <widget class="QToolButton" name="resetBatteryButton">
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="../../imageresources.qrc">
                <normaloff>:/arrow_undo</normaloff>:/arrow_undo</iconset>
              </property>
             </widget>
            </item>
            <item row="0" column="4">
             <spacer name="horizontalSpacer_3">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item row="0" column="1">
             <widget class="QLineEdit" name="batteryEdit">
              <property name="sizePolicy">
               <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
                <horstretch>100</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>100</width>
                <height>0</height>
               </size>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="temperatureHeaderLabel">
              <property name="text">
               <string>Tempertature:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QLineEdit" name="temperatureEdit">
              <property name="sizePolicy">
               <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
                <horstretch>100</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>100</width>
                <height>0</height>
               </size>
              </property>
             </widget>
            </item>
            <item row="1" column="3">
             <widget class="QToolButton" name="resetTemperatureButton">
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="../../imageresources.qrc">
                <normaloff>:/arrow_undo</normaloff>:/arrow_undo</iconset>
              </property>
             </widget>
            </item>
            <item row="1" column="4">
             <spacer name="horizontalSpacer_4">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>256</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="widget_11" native="true">
           <layout class="QHBoxLayout" name="horizontalLayout_6">
            <property name="margin">
             <number>0</number>
            </property>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>AttachableFrame</class>
   <extends>QWidget</extends>
   <header>attachableframe.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../imageresources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>gameTimeEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>SoccerControlFrame</receiver>
   <slot>editGameTime()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>95</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>-4</x>
     <y>31</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>gameTimeResetButton</sender>
   <signal>clicked()</signal>
   <receiver>SoccerControlFrame</receiver>
   <slot>resetGameTime()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>205</x>
     <y>45</y>
    </hint>
    <hint type="destinationlabel">
     <x>301</x>
     <y>30</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>team1GoalsEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>SoccerControlFrame</receiver>
   <slot>editTeam1Goals()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>80</x>
     <y>103</y>
    </hint>
    <hint type="destinationlabel">
     <x>-7</x>
     <y>85</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>team2GoalsEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>SoccerControlFrame</receiver>
   <slot>editTeam2Goals()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>242</x>
     <y>108</y>
    </hint>
    <hint type="destinationlabel">
     <x>302</x>
     <y>91</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>team1PenaltiesEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>SoccerControlFrame</receiver>
   <slot>editTeam1Penalties()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>94</x>
     <y>126</y>
    </hint>
    <hint type="destinationlabel">
     <x>-3</x>
     <y>113</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>team2PenaltiesEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>SoccerControlFrame</receiver>
   <slot>editTeam2Penalties()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>265</x>
     <y>118</y>
    </hint>
    <hint type="destinationlabel">
     <x>300</x>
     <y>118</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playModeComboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>SoccerControlFrame</receiver>
   <slot>selectPlayMode(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>104</x>
     <y>171</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playModeButton</sender>
   <signal>clicked()</signal>
   <receiver>SoccerControlFrame</receiver>
   <slot>changePlayMode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>264</x>
     <y>183</y>
    </hint>
    <hint type="destinationlabel">
     <x>301</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>kickOffButton</sender>
   <signal>clicked(bool)</signal>
   <receiver>SoccerControlFrame</receiver>
   <slot>kickOff()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>32</x>
     <y>216</y>
    </hint>
    <hint type="destinationlabel">
     <x>-5</x>
     <y>204</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playersListWidget</sender>
   <signal>currentRowChanged(int)</signal>
   <receiver>SoccerControlFrame</receiver>
   <slot>selectPlayer(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>40</x>
     <y>351</y>
    </hint>
    <hint type="destinationlabel">
     <x>-3</x>
     <y>272</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>batteryEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>SoccerControlFrame</receiver>
   <slot>editBattery()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>125</x>
     <y>503</y>
    </hint>
    <hint type="destinationlabel">
     <x>-2</x>
     <y>485</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>resetBatteryButton</sender>
   <signal>clicked()</signal>
   <receiver>SoccerControlFrame</receiver>
   <slot>resetBattery()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>207</x>
     <y>503</y>
    </hint>
    <hint type="destinationlabel">
     <x>298</x>
     <y>490</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>temperatureEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>SoccerControlFrame</receiver>
   <slot>editTemperature()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>106</x>
     <y>531</y>
    </hint>
    <hint type="destinationlabel">
     <x>-3</x>
     <y>525</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>resetTemperatureButton</sender>
   <signal>clicked()</signal>
   <receiver>SoccerControlFrame</receiver>
   <slot>resetTemperature()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>211</x>
     <y>531</y>
    </hint>
    <hint type="destinationlabel">
     <x>301</x>
     <y>527</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>editGameTime()</slot>
  <slot>editTeam1Goals()</slot>
  <slot>editTeam2Goals()</slot>
  <slot>editTeam1Penalties()</slot>
  <slot>editTeam2Penalties()</slot>
  <slot>selectPlayMode(int)</slot>
  <slot>changePlayMode()</slot>
  <slot>kickOff()</slot>
  <slot>clearPlayers()</slot>
  <slot>selectPlayer(int)</slot>
  <slot>editBattery()</slot>
  <slot>editTemperature()</slot>
  <slot>resetGameTime()</slot>
  <slot>resetBattery()</slot>
  <slot>resetTemperature()</slot>
 </slots>
</ui>
